### PR TITLE
Feat/e2e example

### PR DESCRIPTION
**Pull request by mefellows**
E2E working JS example, demonstrating the following Pact features:

* Consumer testing and pact file generation, including advanced features like:
    * [Flexible matching](https://docs.pact.io/documentation/javascript/flexible_matching.html)
    * [Provider states](https://docs.pact.io/documentation/provider_states.html)
* Sharing Pacts by publishing to and retrieving from a [Pact Broker](https://github.com/bethesque/pact_broker)
* Provider side verification

This comprises a complete E2E example that can be used as a basis for projects.

### CODE REVIEW

File: examples/e2e/.eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+rules:
```
Comment (tarciosaraiva) [inline examples/e2e/.eslintrc.yml]: so, by default we use [StandardJS](https://github.com/feross/standard)... this filed should not be needed.

File: examples/e2e/consumer.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const express = require('express');
+const request = require('superagent-bluebird-promise');
```
Comment (tarciosaraiva) [inline examples/e2e/consumer.js]: you don't need that... Superagent 2 already has promises embedded into it. Unless you want to use something specific from Bluebird this is overkill

File: examples/e2e/consumerService.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+let { server } = require('./consumer.js');
+
+server = server.listen(8080, () => {
```
Comment (tarciosaraiva) [inline examples/e2e/consumerService.js]: this association is kinda overkill, not sure why you would need that

Comment (mefellows) [inline examples/e2e/consumerService.js]: You can see that it's used below in outputting which host/ip it's bound to, but i'll simplify it.

File: examples/e2e/consumer.js (hunk 2 above)
Comment (mefellows) [inline examples/e2e/consumer.js]: Cool, will fix it. Don't need anything bluebird specific.

File: examples/e2e/.eslintrc.yml (hunk 1 above)
Comment (mefellows) [inline examples/e2e/.eslintrc.yml]: So just remove all rules and inherit from recommended? OK

**Comment by mefellows** [comment]
Implemented feedback, thanks @tarciosaraiva - merging!

**Pull request merged**